ESM module resolution error: Cannot find module './utils/logger'

## Bug Description

Running the compiled server with `node dist/index.js` fails with a module resolution error:

```
Error [ERR_MODULE_NOT_FOUND]: Cannot find module '/path/to/imessage-mcp/dist/utils/logger' imported from /path/to/imessage-mcp/dist/index.js
```

## Root Cause

The project uses ES modules (`"type": "module"` in package.json) but the TypeScript configuration uses `"moduleResolution": "Bundler"`, which doesn't add `.js` extensions to compiled imports.

When running directly on Node.js (not through a bundler), ESM requires explicit `.js` extensions for all relative imports.

## Fix

1. Update `tsconfig.json`:
```json
{
  "compilerOptions": {
    "module": "NodeNext",
    "moduleResolution": "NodeNext",
    // ... rest of config
  }
}
```

2. Add `.js` extensions to all local imports in source files:
```typescript
// Before
import { logger } from "./utils/logger";

// After
import { logger } from "./utils/logger.js";
```

### Files requiring import updates:
- `src/index.ts`
- `src/utils/sdk.ts`
- `src/utils/contacts.ts`
- `src/tools/index.ts`
- `src/tools/messages.ts`
- `src/tools/conversations.ts`
- `src/tools/attachments.ts`
- `src/test.ts`

## Environment
- Node.js v20.19.5
- macOS

Happy to submit a PR with the fix if helpful!

--- a/src/index.ts
+++ b/src/index.ts
@@ -2,9 +2,9 @@ import { McpServer } from "@modelcontextprotocol/sdk/server/mcp.js";
 import { StdioServerTransport } from "@modelcontextprotocol/sdk/server/stdio.js";
 import "dotenv/config";
 
-import { logger } from "./utils/logger";
-import { initializeSDK, closeSDK } from "./utils/sdk";
-import { registerAllTools } from "./tools";
+import { logger } from "./utils/logger.js";
+import { initializeSDK, closeSDK } from "./utils/sdk.js";
+import { registerAllTools } from "./tools/index.js";
 
 // Set log level from environment
 if (process.env.LOG_LEVEL) {

--- a/src/test.ts
+++ b/src/test.ts
@@ -8,7 +8,7 @@ import {
   getAllContacts,
   searchContacts,
   findContactByPhone,
-} from "./utils/contacts";
+} from "./utils/contacts.js";
 
 const sdk = new IMessageSDK({
   debug: false,

--- a/src/tools/attachments.ts
+++ b/src/tools/attachments.ts
@@ -1,7 +1,7 @@
 import { z } from "zod";
 import { McpServer } from "@modelcontextprotocol/sdk/server/mcp.js";
-import { getSDK } from "../utils/sdk";
-import { logger } from "../utils/logger";
+import { getSDK } from "../utils/sdk.js";
+import { logger } from "../utils/logger.js";
 
 export function registerAttachmentTools(server: McpServer) {
   // Get messages with attachments

--- a/src/tools/conversations.ts
+++ b/src/tools/conversations.ts
@@ -1,8 +1,8 @@
 import { z } from "zod";
 import { McpServer } from "@modelcontextprotocol/sdk/server/mcp.js";
-import { getSDK } from "../utils/sdk";
-import { logger } from "../utils/logger";
-import { searchContacts, getAllContacts } from "../utils/contacts";
+import { getSDK } from "../utils/sdk.js";
+import { logger } from "../utils/logger.js";
+import { searchContacts, getAllContacts } from "../utils/contacts.js";
 
 export function registerConversationTools(server: McpServer) {
   // Find contact by name - searches both iMessage chats AND macOS Contacts

--- a/src/tools/index.ts
+++ b/src/tools/index.ts
@@ -1,8 +1,8 @@
 import { McpServer } from "@modelcontextprotocol/sdk/server/mcp.js";
-import { registerMessageTools } from "./messages";
-import { registerConversationTools } from "./conversations";
-import { registerAttachmentTools } from "./attachments";
-import { logger } from "../utils/logger";
+import { registerMessageTools } from "./messages.js";
+import { registerConversationTools } from "./conversations.js";
+import { registerAttachmentTools } from "./attachments.js";
+import { logger } from "../utils/logger.js";
 
 export function registerAllTools(server: McpServer) {
   logger.info("Registering MCP tools...");

--- a/src/tools/messages.ts
+++ b/src/tools/messages.ts
@@ -1,7 +1,7 @@
 import { z } from "zod";
 import { McpServer } from "@modelcontextprotocol/sdk/server/mcp.js";
-import { getSDK } from "../utils/sdk";
-import { logger } from "../utils/logger";
+import { getSDK } from "../utils/sdk.js";
+import { logger } from "../utils/logger.js";
 
 export function registerMessageTools(server: McpServer) {
   // Get messages with filtering

--- a/src/utils/contacts.ts
+++ b/src/utils/contacts.ts
@@ -7,7 +7,7 @@ import Database from "better-sqlite3";
 import { existsSync, readdirSync } from "fs";
 import { homedir } from "os";
 import { join } from "path";
-import { logger } from "./logger";
+import { logger } from "./logger.js";
 
 export interface Contact {
   id: number;

--- a/src/utils/sdk.ts
+++ b/src/utils/sdk.ts
@@ -1,5 +1,5 @@
 import { IMessageSDK } from "@photon-ai/imessage-kit";
-import { logger } from "./logger";
+import { logger } from "./logger.js";
 
 let sdk: IMessageSDK | null = null;
 

--- a/tsconfig.json
+++ b/tsconfig.json
@@ -1,8 +1,8 @@
 {
   "compilerOptions": {
     "target": "ES2022",
-    "module": "ESNext",
-    "moduleResolution": "Bundler",
+    "module": "NodeNext",
+    "moduleResolution": "NodeNext",
     "lib": ["ES2022"],
     "outDir": "./dist",
     "rootDir": "./src",

